x-pack/plugins/lens/public/types.ts

```diff
@@ -38,8 +38,11 @@ export interface DatasourceSuggestion<T = unknown> {
  * Interface for the datasource registry
  */
 export interface Datasource<T = unknown> {
-  // For initializing from saved object
-  initialize: (state?: T) => Promise<T>;
+  // For initializing, either from an empty state or from persisted state
+  initialize: (state?: T | any) => Promise<T>;
+
+  // Given the current state, which parts should be saved?
+  getPersistedState: (state: T) => object;
```

Comment:
Same here (this is also addressed in my stacked PR) - we can do better than just `object` for the part of the state which gets persisted.

Plus I would like to rename this `getPersistableState`, since isn't persisted yet as the datasource doesn't take care of this itself.

Reply:
Agreed